### PR TITLE
feat(plasmantine): Create the modal component with Header

### PR DESCRIPTION
**Pull request by paulgerold**
### Proposed Changes

[CI-191](https://coveord.atlassian.net/browse/CI-191)

Create the modal component with Header

### Potential Breaking Changes

none

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [x] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by gdostie** [comment]
Do you mind adding a demo page for it? 🙏

**Comment by paulgerold** [comment]
> Do you mind adding a demo page for it? 🙏

Done

File: packages/mantine/src/components/modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export interface CustomModalProps extends Omit<ModalProps, 'centered' | 'title'> {
+    /**
+     * Header component props (layout/Header)
+     */
+    header: HeaderProps;
+}
+
+export const Modal: FunctionComponent<CustomModalProps> = ({
```
Comment (gdostie) [inline packages/mantine/src/components/modal/Modal.tsx:16]: Don't forget to add api styles and theming support https://mantine.dev/guides/custom-components/

**Review by gdostie — COMMENTED** [review]
I had more something looking like this in mind when thinking about using the Header component inside the modal title

image

Here is the code that used to do it
```tsx
export interface CustomModalProps extends ModalProps {
    /**
     * Header component props (layout/Header)
     */
    header?: Omit<HeaderProps, 'children' | 'description'>;
    description?: HeaderProps['description'];
}

export const Modal: FunctionComponent<CustomModalProps> = ({children, header, title, description, ...otherProps}) => (
    <MantineModal
        {...otherProps}
        styles={{title: {width: '100%'}}}
        title={
            <Header py={0} px={0} description={description} {...header}>
                {title}
            </Header>
        }
    >
        {children}
    </MantineModal>
);
```

Demo code:
```tsx
<>
    <Modal
        opened={opened}
        onClose={() => setOpened(false)}
        title="Modal title"
        description="Modal description"
        header={{
            actions: (
                <>
                    <Button>Action 1</Button>
                    <Button variant="outline">Action 2</Button>
                </>
            ),
        }}
    >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam ut dui sed sapien finibus malesuada id
        sit amet risus. Praesent finibus sapien vel dolor bibendum, eget euismod metus dignissim. Phasellus
        lacinia sem nunc, vel dapibus odio suscipit id. Aenean lobortis sollicitudin suscipit. Cras vitae ipsum
        sit amet nibh efficitur imperdiet. Praesent scelerisque erat est. Cras dictum sodales tellus sed pretium
    </Modal>
    <Button onClick={() => setOpened(true)}>Open Modal</Button>
</>
```

**Review by gdostie — APPROVED** [review]

**Pull request merged**